src/AngleInterval.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * LSST Data Management System
+ * Copyright 2014-2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_SG_ANGLEINTERVAL_H_
+#define LSST_SG_ANGLEINTERVAL_H_
+
+/// \file
+/// \author Serge Monkewitz
+/// \brief This file defines a class for representing angle intervals.
+
+#include "Angle.h"
+#include "Interval.h"
+
+
+namespace lsst {
+namespace sg {
+
+/// `AngleInterval` is closed interval of arbitrary angles.
+class AngleInterval : public Interval<AngleInterval, Angle> {
+    typedef Interval<AngleInterval, Angle> Base;
+public:
+    // Factory functions
+    static AngleInterval fromDegrees(double x, double y) {
+        return AngleInterval(Angle::fromDegrees(x),
+                             Angle::fromDegrees(y));
+    }
+    static AngleInterval fromRadians(double x, double y) {
+        return AngleInterval(Angle::fromRadians(x),
+                             Angle::fromRadians(y));
+    }
+    static AngleInterval empty() {
+        return AngleInterval();
+    }
+    static AngleInterval full() {
+        return AngleInterval(Angle(-std::numeric_limits<double>::infinity()),
+                             Angle(std::numeric_limits<double>::infinity()));
+    }
+
+    AngleInterval() : Base() {}
+
+    explicit AngleInterval(Angle x) : Base(x) {}
+
+    AngleInterval(Angle x, Angle y) : Base(x, y) {}
+
+    AngleInterval(Base const & base) : Base(base) {}
+};
```

Comment:
I think this class essentially requires the `Angle`s its constructed with to be normalized (or be un-normalized in a consistent way, if that makes any sense) in order for any the topological operations it inherits from `Interval` to work.  And even if the inputs are normalized, I don't think it necessarily handles intervals that cross the wrapping point correctly.

To me, that suggests that `AngleInterval` should normalize its inputs on construction (selectively, of course - the upper bound should never be normalized to a value below the lower bound), and users should not be allowed to modify its internals directly (which would of course mean you couldn't allow that for other `Interval`s either, if you want to preserve the inheritance).